src/SWP/Bundle/ContentBundle/Resources/config/services.yml

```diff
@@ -107,3 +108,27 @@ services:
         parent: "router.symfony"
         tags:
             - { name: router, priority: 300 }
+
+    swp_content_bundle.valid_route_id:
+        class: SWP\Bundle\ContentBundle\Validator\Constraints\RouteIdValidator
+        arguments:
+            - '@swp.provider.route'
+        tags:
+            - { name: validator.constraint_validator, alias: route_in_database }
+
+    swp_content_bundle.routetoarticle.repository:
```

Comment:
why not using storage bundle to register all needed resources? you define the configuration class, register all services, repositories, factories, object managers by calling one method and you will have a simple configuration, where you will be able to change repository, model class if needed etc., you wont have to register any services in yaml config and you wont have to modify the code.